homeassistant/components/flexmeasures/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""The FlexMeasures integration."""
+from __future__ import annotations
+
+import logging
+
+from flexmeasures_client import FlexMeasuresClient
+import isodate
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+# from .api import S2FlexMeasuresClient, async_register_s2_api
+from .const import DOMAIN
+from .helpers import get_from_option_or_config
+from .services import async_setup_services, async_unload_services
+from .websockets import WebsocketAPIView
+
+ATTR_NAME = "name"
+DEFAULT_NAME = "World"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up FlexMeasures from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = entry
+    hass.data[DOMAIN]["current_id"] = entry.entry_id
+
+    config_data = dict(entry.data)
+    # Registers update listener to update config entry when options are updated.
+    unsub_options_update_listener = entry.add_update_listener(options_update_listener)
+    # Store a reference to the unsubscribe function to cleanup if an entry is unloaded.
+    config_data["unsub_options_update_listener"] = unsub_options_update_listener
+    client = FlexMeasuresClient(
+        host=config_data["host"],
+        email=config_data["username"],
+        password=config_data["password"],
+        session=async_get_clientsession(hass),
+    )
+
+    # store config
+    hass.data[DOMAIN]["frbc_config"] = {
```

Comment:
These could be constants

Reply:
changed to constant: 34198d314c5750a8c545234375076f33fe346e83